satpy/composites/lightning.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Composite classes for the LI instrument."""
+
+import logging
+import sys
+
+import numpy as np
+import xarray as xr
+
+from satpy.composites import CompositeBase
+
+LOG = logging.getLogger(__name__)
+
+
+class LightningTimeCompositor(CompositeBase):
+      """Class used to create the flash_age compositor usefull for lighting event visualisation.
+
+      The datas used are dates related to the lightning event that should be normalised between
+      0 and 1. The value 1 corresponds to the latest lightning event and the value 0 corresponds
+      to the latest lightning event - time_range. The time_range is defined in the satpy/etc/composites/li.yaml
+      and is in minutes.
+      """
+      def __init__(self, name, prerequisites=None, optional_prerequisites=None, **kwargs):
+          """Initialisation of the class."""
+          self.name = name
+          super().__init__(name, prerequisites, optional_prerequisites, **kwargs)
+          # Get the time_range which is in minute
+          self.time_range = self.attrs["time_range"]
+          self.standard_name = self.attrs["standard_name"]
+          self.reference_time = self.attrs["reference_time"]
+
+
+      def _normalize_time(self,data:xr.DataArray,attrs:dict)->xr.DataArray:
+          """Normalised the time in the range between [end_time,end_time - time_range].
```

Comment:
```suggestion
      def _normalize_time(self, data:xr.DataArray,attrs:dict) -> xr.DataArray:
          """Normalize the time in the range between [end_time, end_time - time_range].
```

I'm not sure why the linter isn't marking the spaces or the docstring first word. Ruff must not be able to do that yet.